src/backend/gencomm.py

```diff
@@ -96,7 +97,10 @@ def switcher_builtin(backend, OUTFILE, blklens, builtin_types, builtin_maps, typ
     yutils.display(OUTFILE, "switch (%s->u.builtin.handle) {\n" % child_type_str(typelist))
 
     for b in builtin_types:
-        switcher_builtin_element(backend, OUTFILE, blklens, typelist, pupstr, "YAKSA_TYPE__%s" % b.replace(" ", "_"), b.replace(" ", "_"))
+        if b == "bool":
+            switcher_builtin_element(backend, OUTFILE, blklens, typelist, pupstr, "YAKSA_TYPE___BOOL", "bool")
+        else:
+            switcher_builtin_element(backend, OUTFILE, blklens, typelist, pupstr, "YAKSA_TYPE__%s" % b.replace(" ", "_"), b.replace(" ", "_"))
```

Comment:
Instead of add another entry for `bool`, can we do this instead:
```
if b=='_Bool' and backend=="hip":
    ...
else:
    ...
```
?

Reply:
Just making that change is not sufficient as hip does not recognize '_Bool'. We need to use `bool` in `src/backend/hip/genpup.py`. We need to map `_Bool` to `bool` in src/backend/hip/genpup.py`.

Reply:
I see, probably we need to do a replacement in a different place. Let's proceed to merge this. We can do another patch once we understand more.